test/over_react/component/error_boundary_test.dart

```diff
@@ -716,6 +720,215 @@ void main() {
       });
     });
 
+    group('logs errors using a `logger`', () {
+      List<Map<String, List>> calls;
+      List<LogRecord> logRecords;
+      const identicalErrorFrequencyToleranceInMs = 500;
+
+      // Cause an error to be thrown within a ReactJS lifecycle method
+      void triggerAComponentError() {
+        queryByTestId(jacket.getInstance(), 'flawedComponent_flawedButton').click();
+      }
+
+      void sharedSetup({String loggerName, bool shouldLogErrors = true, Logger customLogger}) {
+        calls = [];
+        jacket = mount(
+          (ErrorBoundary()
+            ..loggerName = loggerName
+            ..shouldLogErrors = shouldLogErrors
+            ..logger = customLogger
+            ..identicalErrorFrequencyTolerance = const Duration(milliseconds: identicalErrorFrequencyToleranceInMs)
+            ..onComponentDidCatch = (err, info) {
+              calls.add({'onComponentDidCatch': [err, info]});
+            }
+            ..onComponentIsUnrecoverable = (err, info) {
+              calls.add({'onComponentIsUnrecoverable': [err, info]});
+            }
+          )(Flawed()()),
+          attachedToDocument: true,
+        );
+
+        logRecords = [];
+        expect(jacket.getDartInstance().loggerName, customLogger?.name ?? loggerName ?? defaultErrorBoundaryLoggerName,
+            reason: 'The loggerName getter should return the name of the logger '
+                    'that will log component errors if/when they happen.');
+        Logger(jacket.getDartInstance().loggerName).onRecord.listen(logRecords.add);
```

Comment:
Listeners are cleared in other setups, but they still hang around at the end of at least one test, which is not good.

Proper cleanup is pretty easy to do with the subscription and `addTearDown`:
```suggestion
        final subscription = Logger(jacket.getDartInstance().loggerName).onRecord.listen(logRecords.add);
        addTearDown(subscription.cancel);
```
and then you don't need `clearListeners` anywhere!